### PR TITLE
Issues/91 new cells

### PR DESCRIPTION
**Pull request by aerych**
Refs #91 

This PR is a first pass at cell UI for stories, text notes and photos, and stubbed cells for video and audio notes.

There is a lot of boiler plate in the changes (and project file and xib changes). Some highlights are:
- A CGSize extension for Hashable conformance.
- An ImageResizer utility used to downsize UIImages for display in cells.
- Thumbnail handling in the AssetsViewController
- The cells and cellprovider pattern (for decoupling the StoryAsset model from cells).
- StoryFolder helper methods and StoryAsset provider conformance.

Note: The behavior of the "cloud-upload" icons/buttons is not fully implemented. It will be revisited when syncing is addressed.  Also, video and audio cells are stubs and will see some modifications as video and audio features are implemented, so we'll skip the visual review for now.  Everything will get a final future design review closer to the v1 wrap up.

To test:
- Prep: Go to the Photos app. Select some photos and copy them.  Then use the files app to paste the photos in your current story folder.  This is to work around the known issue of imported images being all black, and will let you see the thumbnail images clearly. 
- Add some text notes and photos.
- Confirm the stories cell reflects the number of assets by asset type.
- Ensure cell layout looks good. (Nothing obviously broken.)
- Confirm cell colors look good in both light and dark mode.
- On the iPad, confirm that cells respect readable width margins.

@jleandroperez could I trouble you for a review of this one?  I apologize for the size, but about half of the changes are the xib and project file. 😬 

Examples:

Screen Shot 2020-08-18 at 11 29 07 AM
Screen Shot 2020-08-18 at 11 29 23 AM
Screen Shot 2020-08-18 at 11 30 06 AM
Screen Shot 2020-08-18 at 11 29 40 AM
Screen Shot 2020-08-18 at 11 30 13 AM
Screen Shot 2020-08-18 at 11 29 54 AM

### CODE REVIEW

File: Newspack/Newspack/Controllers/AssetsViewController.swift [hunk 1]
```diff
@@ -123,14 +130,66 @@ extension AssetsViewController {
...
+            return configurePhotoCell(tableView: tableView, indexPath: indexPath, storyAsset: storyAsset)
+        case .video:
+            return configureVideoCell(tableView: tableView, indexPath: indexPath, storyAsset: storyAsset)
+        case .audioNote:
+            return configureAudioCell(tableView: tableView, indexPath: indexPath, storyAsset: storyAsset)
+        }
+    }
+
+    func configureTextCell(tableView: UITableView, indexPath: IndexPath, storyAsset: StoryAsset) -> TextNoteTableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: TextNoteTableViewCell.reuseIdentifier, for: indexPath) as! TextNoteTableViewCell
```
Comment (jleandroperez) [inline Newspack/Newspack/Controllers/AssetsViewController.swift]: I'm selling [this extension API](https://github.com/Automattic/simplenote-ios/blob/develop/Simplenote/Classes/UITableView+Simplenote.swift#L35) REALLY REALLY cheap 😂

File: Newspack/Newspack/Controllers/AssetsViewController.swift [hunk 2]
```diff
@@ -123,14 +130,66 @@ extension AssetsViewController {
...
+        let folderManager = SessionManager.shared.folderManager
+        guard
+            let bookmark = asset.bookmark,
+            let url = folderManager.urlFromBookmark(bookmark: bookmark),
+            let image = UIImage(contentsOfFile: url.path)
+        else {
+            return nil
+        }
+
+        return ImageResizer.shared.resizeImage(image: image, identifier: asset.uuid.uuidString, fillingSize: size)
```
Comment (jleandroperez) [inline Newspack/Newspack/Controllers/AssetsViewController.swift:192]: Should this one be async, or it's an overkill?

File: Newspack/Newspack/Extensions/CGSize+Helpers.swift [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import UIKit
+
+// Helper extension to make CGSize hashable so it can be used as dictionary keys.
+//
+extension CGSize: Hashable {
```
Comment (jleandroperez) [inline Newspack/Newspack/Extensions/CGSize+Helpers.swift:5]: Neeeat!

File: Newspack/Newspack/Models/StoryAsset+CoreDataClass.swift [hunk 4]
```diff
@@ -2,15 +2,18 @@ import Foundation
 import CoreData
 
 @objc(StoryAsset)
-public class StoryAsset: NSManagedObject {
+public class StoryAsset: NSManagedObject, TextNoteCellProvider, PhotoCellProvider, VideoCellProvider, AudioCellProvider {
+
+    var caption: String! {
```
Comment (jleandroperez) [inline Newspack/Newspack/Models/StoryAsset+CoreDataClass.swift:7]: Maybe drop the `!` ?

**Review by jleandroperez — APPROVED** [review]
✅ Code looks square
✅ Stories reflect the number of assets they contain
✅ Cell layout looks good
✅ Cell colors are great as well!
✅ iPad Margins are on spot

One tiny detail I've noticed is: iPad + Light Mode, nearly all of the navigation controls are not visible.
Attaching a screenshot below!

**However** this PR didn't introduce those controls, so definitely good to :shipit: (100% up to you sir!!)

File: Newspack/Newspack/Controllers/AssetsViewController.swift (hunk 2 above)
Comment (aerych) [reply Newspack/Newspack/Controllers/AssetsViewController.swift:192]: I'm thinking overkill based on what I've read, but I haven't done any stress testing yet.  Switching to async would be a must if there is a performance issue but I'm content to leave it for now vs over-optimize too early.

**Comment by aerych** [comment]
> iPad + Light Mode, nearly all of the navigation controls are not visible

This is actually due to the authenticator's navbar and bar button styles being a bit too broadly applied and is resolved in https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/371.  The change will be reflected in NiOS the next time I update the auth pod version. :)

**Comment by jleandroperez** [comment]
Ahhh gotcha, thank you sir!!

File: Newspack/Newspack/Models/StoryAsset+CoreDataClass.swift (hunk 4 above)
Comment (aerych) [reply Newspack/Newspack/Models/StoryAsset+CoreDataClass.swift:7]: Hmm...yeah.  I did that for consistency with other provider properties that basically just reference the StoryAsset properties. I'd really prefer getting rid of the optionals altogether. I'll give this some more thought and revisit.

File: Newspack/Newspack/Controllers/AssetsViewController.swift (hunk 1 above)
Comment (aerych) [reply Newspack/Newspack/Controllers/AssetsViewController.swift]: Nice! I'm totally buying :D

**Review by jleandroperez — APPROVED** [review]
:shipit: stands sir! 🎉

**Comment by aerych** [comment]
Thank you Jorge!!

**Pull request merged**